lib/ReactViews/Map/Legend/LocationBar.tsx

```diff
@@ -0,0 +1,91 @@
+import classNames from "classnames";
+import { observer } from "mobx-react";
+import React, { RefObject, useEffect, useRef } from "react";
+import { useTranslation } from "react-i18next";
+import styled from "styled-components";
+import Terria from "../../../Models/Terria";
+import MouseCoords from "../../../ReactViewModels/MouseCoords";
+import Styles from "./legend.scss";
+
+interface PropsType {
+  terria: Terria;
+  showUtmZone: boolean;
+  mouseCoords: MouseCoords;
+}
+
+const LocationBar: React.FC<PropsType> = observer(({ mouseCoords }) => {
```

Comment:
And now mouseCoords is no longer an observable so should cut out overhead, and still rerender based on props being updated?

Reply:
The update is triggered by a [new event listener](https://github.com/TerriaJS/terriajs/blob/24d5ecb6084ec47b72440bfc5f7edb1be0bf9cac/lib/ReactViews/Map/Legend/LocationBar.tsx#L26). This is much more efficient because it avoids the mobx overhead. I'm also using react refs to update the DOM which [avoids the re-rendering](https://github.com/TerriaJS/terriajs/blob/24d5ecb6084ec47b72440bfc5f7edb1be0bf9cac/lib/ReactViews/Map/Legend/LocationBar.tsx#L27-L32) altogether.